### ISSUE TITLE
Recall procedures for TAG and AB

### ISSUE THREAD

**Issue by mnot** [w3c/process#882]
The Board has a [removal procedure](https://github.com/w3c/board/blob/main/governing-docs/bylaws.md#section-5-resignation-and-removal), but the TAG and AB do not, beyond the CEO's ability to [discipline them](https://www.w3.org/2023/Process-20231103/#discipline).

Given the direct process role that they now play when Councils are convened, it seems appropriate to make the TAG and AB more accountable to the Membership by defining a procedure for recalling them.

Straw-man proposal: a recall can be instigated by a demand signed by 20 Member representatives, and it's executed as a [requisite member vote](https://github.com/w3c/board/blob/main/governing-docs/bylaws.md#section-11-action-by-members).

**Comment by frivoal** [comment]
Only commenting on a side aspect of the proposal for now, rather than on the core question, but invoking Bylaws concepts like _Member representatives_ and _requisite member votes_ for groups like the AB or TAG which are anchored in the Process is weird, and creates inter-dependencies that I'd rather avoid. If anything, we should go with AC Reps.

**Comment by mnot** [comment]
Yes, that bugged me too, but those procedures / definitions seemed more appropriate. Maybe I've just had my head in the bylaws too much lately.

In the longer term, rationalising the bylaws and process should make such things easier (I hope). In the meantime, let's get agreement in principle on the parameters first, and worry about implementation language later.

**Comment by dwsinger** [comment]
one place to look for similarity or grounding might be the appeal procedure in the Process?

**Comment by frivoal** [comment]
> one place to look for similarity or grounding might be the appeal procedure in the Process?

Yes, I was thinking about that too: https://www.w3.org/2023/Process-20231103/#ACAppeal seems like a workable basis. I'm happy to try and draft something based on that. TLDR: it's a two step process, where you need to convince 5% of the voter base that the question is worth taking up to have a vote, and if you do, you hold a vote during which you need to reach majority for the proposal to pass.

Incidentally, both for this recall procedure and for the general AC appeal, I think we could/should borrow the Bylaws's notion of a threshold that varies based on quorum. passing drastic things by simple majority without a quorum requirement worries e a little bit, even if there's not precedent of it being an issue. Maybe I should open a separate issue about that?

**Comment by frivoal** [comment]
See https://github.com/w3c/process/issues/886 for a discussion of adopting voting thresholds along the lines of what the bylaws do for requisite member votes.

**Comment by chaals** [comment]
Addressing the substance, with thoughts that haven't crystallised for me yet:

I'm uneasy about this in general. I understand the need for accountability, but I don't think single-seat recalls provide a very good version of it for a representative body like TAG or AB.

I'd perhaps be OK with a "nuclear spill" option resulting in the whole AB or TAG being put up for election. I'm also leaning to suggesting that we introduce effective good standing requirements for AB and TAG. (I was a proponent in removing them from working groups, where they had become rarely-used to the point that enforcing them was likely unfair in practice).

There's already a lot of asymmetry in the way W3C works, given it already institutionalises _machtpolitik_ by putting a premium on mass adoption in an ecosystem where there are a few superpowers. I am not very concerned about those superpowers behaving badly. But I am concerned that we'll end up motivating others to do so. 

Being able to recall a single individual from the fairly small AB/TAG, presumably on the basis of supporting a position that a small proportion of members strongly disagree with, seems like a tool that invites gaming, and something that makes it less attractive for good-faith actors to stand in the first place, and places a higher price on them acting as they see fit.

IMHO (and I realise people disagree with me) we mitigate some of the asymmetry through the use of STV to elect these increasingly powerful councillors, and by holding consensus as a core value for those bodies. Lowering the barriers to enable decisions to be forced more easily, instead of having to slog through more consensus-building, seems to me a step in the wrong direction.

I'm trying to think through controversial things W3C has done (or decided against), and how a recall procedure would have affected the outcome. My initial guesses suggest more political game-playing and resultant rancour, for no clear practical benefit.

I'd like to see some arguments that explain how this would work in practice and how that would be an overall improvement.

**Comment by dwsinger** [comment]
I tend to agree with Chaals. 

Also the impact on the person concerned of even having a recall vote issued against them, let alone having it succeed, could be a large mark on their career. I have a hard time imagining it would or should ever be used.

I suppose we could have a "vote of no confidence" in the TAG or AB as a whole, which triggers a "general election", but I cannot imagine a time in the past where that would or could have been used (and the efficacy of both bodies has gone up and down over the years).

**Comment by cwilso** [comment]
I'm somewhat ambivalent about recall procedures; I think they are a nuclear option that are unlikely to be applied, or at least applied repeatedly or without serious reason to be applied.   I wouldn't object to any of the options suggested.

My disagreement with you, Chaals, is that STV is increasingly excluding those "superpowers" (as it prioritizes diversity of opinion, those affiliated with centrist "superpowers" are unlikely to garner enough votes to get elected), and that increase in diverse representation has sadly turned into consensus-blocking in practice.  (I.e., significant lack of doers, significant surplus of objectors).  This has diminished the value of those bodies, in my opinion.

As for the impact on someone of having a recall vote issued against them, frankly we do not expose enough real information about participation to initiate recalls on anything but political grounds, and as someone probably more likely to have such a recall issued against me, I don't see it as any more embarrassing or career-marking than our election mechanisms already are, so I don't think it's of great concern.

I will note as I did originally that the "discipline" the CEO has at their disposal already includes the power to remove AB and TAG participants:

> The [CEO](https://www.w3.org/policies/process/20231103/#ceo) may take disciplinary action, including suspending or removing for cause a participant in any group (including the [AB](https://www.w3.org/policies/process/20231103/#advisory-board) and [TAG](https://www.w3.org/policies/process/20231103/#technical-architecture-group)) if serious and/or repeated violations, such as failure to meet the requirements on individual behavior of (a) this process and in particular the CEPC, or (b) the membership agreement, or (c) applicable laws, occur. Refer to the [Guidelines to suspend or remove participants from groups](https://www.w3.org/Guide/process/banning.html).

**Comment by frivoal** [comment]
Personally:
* As a current member of an elected body (the AB), it am comfortable with recall procedures existing. If used without just cause, they're more likely to mar the reputation of the person using them than that of the target, and if used in a case where it would be warranted, it's good to have them.
* As a somewhat long time participant in W3C, I do not recall a single time where I thought a member of the TAG or the AB did something that would warrant a recall, nor a time where I though either body as a whole needed recalling.

I don't have a strong view on individual vs collective recall. Collective is likely less prone to abuse, but also fairly blunt.

**Comment by chaals** [comment]
TL;DR: I've thought a fair bit more. I'm more opposed to individual recall, more supportive of Good Standing requirements but with caveats about how they are  implemented, and ambivalent about total spill mechanisms that would re-elect an entire body.

### Recall

I've spent a collective decade+ on the AB, and have been involved with W3C since before the AB came into existence (and ran for the first TAG) seen situations where I could easily imagine a recall. More importantly IMHO, I can think of many situations where someone on the AB or TAG would have been reluctant to take a principled position given the threat of a recall. Some of those positions may have been frustrating - which is how a recall would be justified - but some I think were actually important.

The way W3C seems to work, if there is someone prepared to take the opprobrium for starting an effort as divisive as a recall, then it suddenly becomes normalised behaviour - may who would never step up and put their name on the initial call will justify supporting it. There are hundreds of members, any of whom can decide to take that initial step on, for any number of reasons.

Finally, although we have a reasonable representation of the membership's diversity of views on those bodies, they are relatively finely balanced. I think the impact of having a recall procedure for individual members, whether used or not, is not helpful.

### Good Standing

Recently the AB has been trying to address the problem that sometimes members of these important bodies who just don't respond.

As I noted above, the bodies are elected in a way that makes them somewhat representative of the diversity of W3C membership. A consequence of this is that the smaller the pool for any given election (whether that's from the split-cycle we still have as a hangover from the way we used to elect people, or the more extreme case of a single vacancy) the more the overall results can be expected merely to reinforce the most powerful group (or set of groups for multiple seats), and the less the results are likely to favour candidates with broad support from their "opposition" as well as deep support from those they most closely represent.

W3C used to use good standing very strictly in some groups to determine whether someone was a participant whose dissent affected the measure of "consensus".

I think it makes sense to have a good standing requirement for these elected bodies. However unfair it may seem to a "good individual", I also think there should not be a discretionary mechanism for enforcement - if it applies it should be written in such a way that it automatically applies to any member who fails to maintain good standing.

I also think that an individual vacancy should just be left open, with a special election only taking place if there are three open seats. A major reason for this is that an individual special election although run entirely in good faith can easily produce pretty undemocratic results in the  relevant circumstances.

For the AB, which is already happy to have only 9 or as many as 11 members, clearly the specific number doesn't seem crucial. Likewise the Board doesn't seem to have felt a need to reach its full complement.

### Full Spill

This is a potentially real scenario. There was a concerted and successful effort to replace the entire TAG, some years ago - it took well over a year to execute in part because it had to be done through the course of two election cycles. I don't know if the people involved would have tried to pull a TAG spill if that option were available, but many of them are still around. We could ask them. (It's probably instructive to see who says they were actively part of that...)

If there is a situation where one of these bodies loses the confidence of the membership, it seems reasonable to allow a complete spill. This means that all seats would be vacated, and the entire body is re-elected in one go.

A key difference with a recall is that while in practice many people will blame the problem on individuals in the relevant body, there is no formal mechanism that will historically reinforce the suggestion that it is one person's fault the body failed to fulfil its function of providing consensus-backed guidance.

It's also fairly resistant to gaming: given that the outcome would be a full election, the bodies will be reconstituted as the most broadly representative they can, in principle sacrificing the value of institutional memory that is the apparent rationale for our split election cycles. A situation a body has "gone rogue" would mean this is appreciated by many, across the various differences of technical and governance opinion that sometimes divide us. 

As an effort to entrench or reinforce power for a specific group, a spill is idiotic, and very likely to fail spectacularly with real blowback. The annoyance of this taking place, meaning an extra election - which could be compensated by automatically stating the it replaces the next expected election - and the relevant group not meeting for a couple of months is probably worth the lesson.

Equally, I think we could continue to live OK without enabling this.

**Comment by frivoal** [comment]
https://github.com/w3c/process/pull/986 has been merged. Waiting before closing this issue until we tidy up follow up questions.

**Comment by fantasai** [comment]
The AB and Process CG have discussed this issue on several occasions, and the current distillation has three parts:

- **Should the AB or TAG be able to eject one of its members by a 3/4 vote?** The AB [resolved to add this ability](https://www.w3.org/2025/03/20-ab-minutes.html#ef3e), and the Process CG merged the core text and will work on follow-up refinements. The 3/4 threshold matches the ejection procedures for Board Directors in the Bylaws, and prevents abuse by a lesser majority to eject a member for simply having contrary opinions.
- **Should the AC be able to recall the AB or TAG, essentially triggering a snap election, by a majority vote?** This has [a draft in the CG](https://github.com/w3c/process/pull/888), but still requires AB discussion.
- **Should the CEO's powers of removal be somehow restricted when applied to elected bodies?** This is pending AB discussion as well.

The second two questions are currently scheduled for the AB F2F following the AC meeting. See [corresponding issue in the AB repository](https://github.com/w3c/AB-memberonly/issues/237).

**Comment by dbaron** [comment]
One thought about recall procedures:  I think it's worth considering the relationship between the recall procedure and the election system.  Currently we have an election system that (very roughly) tries to produce a body that is representative of the distribution of membership's first preferences.  So given an election for four open seats, an election could in theory yield a set of four elected candidates who each have the strong support of about a quarter of the membership and the strong opposition of about three quarters of the membership.  It could also (perhaps more realistically) yield three candidates who have the support of 3/4 and the opposition of 1/4, and one candidate who has the support of 1/4 and the opposition of 3/4.

It seems both awkward and potentially quite subject to abuse if combination of the election rules and the recall rules mean that the candidates elected in an election could be immediately recalled by the same voters expressing the same preferences.

One possible abuse scenario relates, I think, to the possibility of a four seat election with two clear groups of voters that are roughly 3/4 and 1/4 of the electorate.  The election system is (I think) designed to elect three candidates from one group and one from the other.  The one candidate elected from the latter group could then be recalled by the 3/4 of the electorate, triggering a one seat election that would then be won by another candidate representing the group that is 3/4 of the electorate.

(A perhaps larger question is then whether that's a problem with the recall rules or a problem with the election rules.)

**Comment by chaals** [comment]
I think @dbaron identifies a problem that a mechanism to recall a single person raises. To phrase his larger question another way:

Should a 3/4 majority of the members be able to deny a 1/4 minority any representation in TAG/AB...? (To be honest, this was the question that led me to push so long for STV, because I believed and still believe the answer is resoundingly NO).

Should this risk be realised in practice, IMHO it would effectively signal that W3C is not a consensus-based body, and runs in the interests of the strongest power bloc.

I am reasonably confident that won't happen. In part because anyone pushing it to happen looks a lot like a deliberate wrecker, and I have faith that the community are not prepared to tolerate that even if a specific instance supports their own interests.

Claiming that EME was the most divisive issue we ever faced, the AB were unanimously OK with working on it (despite some disagreements on the details), whereas somewhere around 1/4 of the membership were pretty appalled. In my opinion the proponents managed to streamroll opposition, including an appeal against the recommendation failing, but *importantly* I don't think the impact was wider than the specific decisions, which ended up being majority-driven because there was a clear inability to reach consensus.

That last point is important. If there were a wider fallout, instead of the episode demonstrating that W3C can deal (usefully and valuably at least, even if imperfectly) with very complex decisions over divisive issues, it would suggest that W3C simply provides "astroturfing" for the most powerful players  in its space.

On the other hand, I have seen the ideas discussed, "war-gamed", and efforts made to see how they can be put into practice. Whether they are successful or not, and whether the outcome is a good thing or not, the process always causes a certain amount of damage.

It seems to me that introducing this lever into our governance will make it tempting for some to try and pull it. Recalling a single obstructive individual might not seem as extreme as calling a full election, but its effect is likely, as @dbaron noted, to genuinely counter the intent of our electoral procedures to provide representation of our diversity.

There are many tools that can be deployed today against one elected person who strongly opposes the direction of the body they are part of. Most of them are far less harmful to the overall consensus-based nature of W3C than a recall. Having been on both sides of concrete discussions in the past, it seems to me that we aren't solving a real problem by introducing recall, but instead setting ourselves up to create a bunch of new problems.

**Comment by frivoal** [comment]
The possibility for a majority to silence a legitimate minority view is why I would be very strongly opposed to recall by simple majority. But that's not what is being proposed. With a supermajority, I feel this is quite safe. Yes, you could be in a situation where >3/4 of the body disagree vehemently with <1/4, but even in such a case, it seems highly unlikely to me that >3/4 of the AB or TAG would chose to eject some of there members over differences of opinions, even stark ones. If even a few have a sense of propriety, the removal would fail, and I expect more than a few to actually have good sense here.

I would not be surprised if every now and then, a few AB or TAG members are irritated enough with a divergent voice that they wish they would just go away. I would be surprised if that ever rose to a super majority over anything other than gross misbehavior.

But if we somehow end up with such a bullying AB or TAG, a general recall would seem in order, which is why I'd also support the ability for the AC as a whole to recall the AB or TAG.

**Comment by dbaron** [comment]
Ah, indeed I had missed that https://github.com/w3c/AB-memberonly/issues/237#issuecomment-2781374569 and https://github.com/w3c/process/issues/882#issuecomment-2781376693 differed from the initial summary in https://github.com/w3c/AB-memberonly/issues/237#issue-2529186878 (which I read first) in that the recall by majority of the AC was now only for the entire body and not an individual member.

**Comment by frivoal** [comment]
I have created a long-lived topic branch to work on this question. As there continues to be discussion in the AB and TAG on this complex topic (not only how to do it, but also whether it's a good idea in the first place), I have backed out the previously merged commits related to this topic from the main branch, and instead put them on the [ab-tag-discipline topic branch](https://github.com/w3c/process/tree/ab-tag-discipline) ([preview](https://www.w3.org/policies/process/drafts/ab-tag-discipline/)). That way, we can discuss and iteratively adopt individual improvements, look at the whole as they integrate, and still separately decide if we want to land the whole thing or not.

All current pending issues and pull requests on this topic can also be found through the ["Topic: AB/TAG discipline"](https://github.com/w3c/process/issues?q=state%3Aopen%20label%3A%22Topic%3A%20AB%2FTAG%20discipline%22) label.
